kedro/extras/datasets/dask/parquet_dataset.py

```diff
@@ -105,8 +107,36 @@ def _load(self) -> dd.DataFrame:
         )
 
     def _save(self, data: dd.DataFrame) -> None:
+        self._process_schema()
         data.to_parquet(self._filepath, storage_options=self.fs_args, **self._save_args)
 
+    def _process_schema(self) -> None:
+        """Processes the schema in the catalog.yml or the API, if provided.
+        This directly supports the Dask specifications for providing a schema
+        when saving to a parquet file.
+
+        If the schema is a dictionary and the values are strings, we assume that they
+        correspond to pyarrow field types.
+
+        Alternatively, a full pyarrow.Schema type may also be provided if desired.
+        """
+        schema = self._save_args.get("schema")
+
+        if isinstance(schema, dict):
+            for col, field in self._save_args["schema"].items():
+                if isinstance(field, str):
+                    assert field.startswith("pa.") or field.startswith("pyarrow.")
+                    self._save_args["schema"][
+                        col
+                    ] = eval(  # nosec pylint: disable=W0123,I0021
+                        field
+                    )
+        elif isinstance(schema, str) and (
+            schema.startswith("pa.schema") or schema.startswith("pyarrow.schema")
+        ):
+            schema = eval(schema)  # nosec pylint: disable=W0123,I0021
```

Comment:
So I completely get why you want to do this, but I think we're unlikely to accept a raw `eval` statement into the core project due to the security risks. Additionally, it would reduce the project's score in the eyes of vendors like [Snyk](https://snyk.io/advisor/python/kedro).

Reply:
@datajoely yeah, I totally understand. I'm using this patch locally for my use case, but I'm also hesitant about this because of the same reasons you pointed out. Just opening this in case someone has better ideas to make this work. Feel free to close anytime as well! Thank you! :D

Reply:
I am resolving this @datajoely since I have implemented something that no longer requires the use of `eval`. Thanks!